src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -63,20 +72,49 @@ def _process_model_before_weight_loading(self, model, **kwargs):
         from compressed_tensors.quantization import apply_quantization_config
 
         ct_quantization_config = self.compressor.quantization_config
-        apply_quantization_config(model, ct_quantization_config, run_compressed=True)
 
-    def _process_model_after_weight_loading(self, model, **kwargs) -> None:
-        pass
+        # Model is quantized, populate state dict with scale and zp
+        # Compressed models need to be loaded as CompressedLinear for unpacking on decompression
+        apply_quantization_config(
+            model, ct_quantization_config, run_compressed=self.run_compressed or self.is_compressed
```

Comment:
```suggestion
            run_compressed = self.is_compressed and self.run_compressed
            model, ct_quantization_config, run_compressed=run_compressed
```

Doesn't the previous logic have the effect of completely ignoring the run_compressed argument if the model is quantized?

Note that in the future, we plan to perform decompression through the `apply_quantization_config` function